test_extract_unique_sensors_01.py

```diff
@@ -0,0 +1,63 @@
+import unittest
+import logging
+from mock import MagicMock
+from mock import patch
+from extract_unique_sensors_01 import extract_all_sensors
+from extract_unique_sensors_01 import splitting_extracted_sensors
+
+
+class TestExtractSensors(unittest.TestCase):
+    """Test class for function test_read_sensors."""
+
+    def test_read_sensors(self):
+        with patch('requests.get') as mock_request:
+            url = "http://lebogang.com"
+            mock_request.return_value.status_code = 200
+            self.assertTrue(url)
+
+        with patch('requests.get') as mock_request:
+            non_existing_url = "http://google.com/fakeurl"
+            mock_request.return_value.status_code = 404
+            self.assertTrue(non_existing_url)
+
+        file_name = 1
+        mock_path = MagicMock()
+        mock_path.Path(file_name).exists().return_value = True
+        self.assertTrue(file_name)
+        with self.assertRaises(AssertionError):
+            self.assertIsInstance(file_name, str)
+            self.assertIsInstance(url, str)
+
+        with self.assertLogs('test_logger', level='DEBUG') as cm:
+            logging.getLogger('test_logger.file_name').debug("file_name exist")
+            logging.getLogger('test_logger.url').debug("URL exist")
+            logging.getLogger('test_logger').error("file_name or url type does exist")
+            self.assertEqual(
+            cm.output, [
+                'DEBUG:test_logger.file_name:file_name exist',
+                'DEBUG:test_logger.url:URL exist',
+                'ERROR:test_logger:file_name or url type does exist']
+            )
```

Comment:
Shouldn't you be testing the `read_sensors` function?
I noticed that it is not imported, and also you are patching the wrong package instead of `urllib` which was used.

Reply:
@mmphego I see I was using the wrong package that's not even there on my script. I fixed it, and imported the function that I am testing. Thank you.